ebos/eclgenericproblem.cc

```diff
@@ -374,7 +374,11 @@ updateNum(const std::string& name, std::vector<T>& numbers)
     unsigned numElems = gridView_.size(/*codim=*/0);
     numbers.resize(numElems);
     for (unsigned elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-        numbers[elemIdx] = static_cast<T>(numData[elemIdx]) - 1;
+        if (numData[elemIdx] > 0) {
+            numbers[elemIdx] = static_cast<T>(numData[elemIdx]) - 1;
+        } else {
+            throw std::runtime_error("zero or negative values provided for region array: " + name);
+        }
```

Comment:
Do we handle the case of one rank/process throwing an exception here, but other ranks continuing?  What if `numData[elemIdx] - 1` is greater than (or equal to) the size of whatever region table it's being used to index?

Reply:
> Do we handle the case of one rank/process throwing an exception here, but other ranks continuing?


I guess not. If handling means that you get a nice error message. 

You will get something like this. 

```
*** An error occurred in MPI_Allreduce
*** reported by process [1191968769,14]
*** on communicator MPI_COMM_WORLD
*** MPI_ERR_TRUNCATE: message truncated
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
```

I think this is align with many user input error that will make the simulator throw. Do you want to add a layer of communication?


> What if numData[elemIdx] - 1 is greater than (or equal to) the size of whatever region table it's being used to index?

I guess that make sense to add a check for that as well. Current code will seg fault if you put in a satnum > number of tables.

Reply:
This error message will happen at a comletely other place in the simulator (because different processes are doing different communications that are interfering. It might also be dependent on the MPI implementation used. I can imagine other implementations deadlocking. So yes, we need to do something about this,

Reply:
> We need to make sure that if there is a problem for one element then all processes will throw.

Yeah, we need better error handling in parallel.  For that matter, we also really need a human readable diagnostic message.  Something referring to aspects of an MPI operation is not usable for the engineer.

Reply:
Something similar to https://github.com/OPM/opm-simulators/blob/ed591239a7371d972f496b36b0e610929efe79aa/opm/simulators/wells/BlackoilWellModel_impl.hpp#L148

Put `OPM_BEGIN_PARALLEL_TRY_CATCH();` before and `OPM_END_PARALLEL_TRY_CATCH(message, ebosSimulator_.gridView().comm());` after loop. We need to make sure that there no other nested ones inside